src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -119,10 +181,36 @@ public void launch() {
             }
         });
 
+
+
+
         kernel.writeEffectiveConfig();
 
         logger.atInfo().setEventType("system-start").addKeyValue("main", kernel.getMain()).log();
         startupAllServices();
+
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
+    private Queue<DeviceIdentityInterface> findProvisioningPlugins() {
+        Queue<DeviceIdentityInterface> provisioningPlugins = new LinkedList<>();
+        EZPlugins ezPlugins = kernel.getContext().get(EZPlugins.class);
+        try {
+            ezPlugins.withCacheDirectory(nucleusPaths.pluginPath());
+            ezPlugins.implementing(DeviceIdentityInterface.class, (c) -> {
+                try {
+                    provisioningPlugins.add(provisioningPluginFactory.getPluginInstance(c));
+                } catch (InstantiationException | IllegalAccessException e) {
+                    logger.atError().kv("Plugin", c.getName()).setCause(e.getCause())
+                            .log("Error instantiating a provisioning plugin");
+                }
+            });
+
+            // ezPlugins.loadCache();
+        } catch (IOException t) {
+            logger.atError().log("Error launching provisioning plugins", t);
```

Comment:
Silent failure, is this a good idea, it will launch nucleus anyway?

Reply:
I have responded to this in other comment below. We can discuss more on slack if you still feel otherwise.